text/0000-cfg-path-version.md

```diff
@@ -0,0 +1,674 @@
+it is clear that it is paths we are talking about because the argument
+is a path and not something else.
+
+While `can_use` is also a strong contender, we reject this option because
+it may imply to the user that only things that you may `use $path;` can
+go in there. Meanwhile, you may `#[cfg(accessible(::foo::MyTrait::my_method))`
+which is *not* possible as `use ::foo::MyTrait::my_method;`. This also applies
+to other associated items and inherent methods as well as `struct` fields.
+
+## `#[cfg(nightly)`
+
+[dbg]: https://crates.io/crates/dbg
+
+One reason for the inclusion of `#[cfg(nightly)]` is that it is useful on its
+own to conditionally compile based on nightly/not as opposed to providing
+an `unstable` feature in `Cargo.toml`. An example of this is provided by the
+[dbg] crate which currently uses [version_check] to provide this automation.
+
+However, as we've argued and demonstrated in the [guide-level-explanation],
+the ability to `#[cfg(nightly)]` really shines when used in conjunction with
+`#[cfg(accessible($path))]`.
+
+### Alternative `#![if_possible_feature(<feature>)]`
+
+As an alternative to `#[cfg_attr(nightly, feature(<feature>))]`
+we could permit the user to write `#![if_possible_feature(<feature>)]`.
+The advantage of this is that it is quite direct with respect to intent.
+However, adding this in terms of `nightly` already has precedent in
+[version_check]. In addition, `nightly` also composes with other flags
+using `any`, `not`, and `all`.
+
+## `#[cfg(version = "..")`
+
+When it comes to `version = ".."`, it is needed to support conditional compilation
+of language features as opposed to library features as previously shown.
+Also, as we've seen, `version = ".."` does not subsume `accessible(..)` but is
+rather a complementary mechanism.
+
+One problem specific to `version = ".."` is that it might get too `rustc` specific.
+It might be difficult for other Rust implementations than `rustc` to work with
+this version numbering as libraries will compile against `rustc`s release
+numbering. However, it is possible for other implementations to follow
+`rustc` in the numbering and what features it provides. This is probably not
+too unreasonable as we can expect `rustc` to be the reference implementation
+and that other ones will probably lag behind. Indeed, this is the experience
+with `GHC` and alternative Haskell compilers.
+
+### The bikeshed - Argument syntax
+
+We have two options with respect to how the `version` flag may be specified:
+
+1. `version = "<semver>"`
+2. `version("<semver>")`
+
+The syntax in 2. is currently an error in `#[cfg(..)]` as you may witness with:
+
+```rust
+// error[E0565]: unsupported literal
+#[cfg(abracadabra("1.27"))] fn bar() {}
+```
+
+We could allow this syntax. However, we have chosen the syntax in 1.
+This with consistency with flags such as `target_feature = "bmi"`.
+Another reason to go with `version = ".."` is that `version("..")`
+looks like a list.
```

Comment:
I agree with `version = "..."` *only if* it's a static comparison check, i.e. it behaves as if the compiler defaults to a `--cfg 'version="1.29.0"'`. If it has *any* dynamic connotation checks, I prefer `version(semver)` (i.e. *without* quotes), to distinguish it and make it look more like a function/macro (just like `any(...)` and `all(...)`).

Reply:
Changed to `version(<semver>)` with your rationale. :)